kubernetes-manifests/cartservice.yaml

```diff
@@ -26,6 +26,12 @@ spec:
         app: cartservice
     spec:
       terminationGracePeriodSeconds: 5
+      initContainers:
```

Comment:
I'm -1 on adding an initContainer. While it fixes the immediate problem, it adds hard dependency to redis to start up. This is not regarded as a good practice in Kubernetes overall. It's a hack.

Instead we can fix cartservice's health method to not to talk to redis.

I think we should solve the root problem here. cartservice is almost always gonna start slower than a redis container (it's larger in size, and uses .net which has a much slower startup time than a C program).

Reply:
@simonz130 thoughts? Can we just remove the health check? (https://github.com/GoogleCloudPlatform/microservices-demo/blob/master/src/cartservice/HealthImpl.cs)

Removing it seems to work fine for me, but I'm not too familiar with C#